src/jit/lclvars.cpp

```diff
@@ -133,14 +133,13 @@ void                Compiler::lvaInitTypeRef()
     // Change the compRetNativeType if we are returning a struct by value in a register
     if (!hasRetBuffArg && varTypeIsStruct(info.compRetNativeType))
     {
-#ifdef _TARGET_ARM_
-        // TODO-ARM64-NYI: HFA
+#if FEATURE_MULTIREG_RET && defined(FEATURE_HFA)
         if (!info.compIsVarArgs && !opts.compUseSoftFP && IsHfa(info.compMethodInfo->args.retTypeClass))
```

Comment:
Another side thing: the right way for softFP to disable HFA calling convention is to define FEATURE_HFA as zero so that both VM and JIT get compiled without HFA support rather than adding runtime checks to disable HFA calling convention.  Post your check-in, you might want to open a Git issue for it.